app/models/spree/address_decorator.rb

```diff
@@ -1,5 +1,5 @@
 Spree::Address.class_eval do
-  has_one :enterprise
+  has_one :enterprise, dependent: :restrict
   belongs_to :country, class_name: "Spree::Country"
```

Comment:
This needs a test. Could be a simple one trying to delete an address used in an enterprise.

Reply:
Hm, I think I agree that a unit test could be useful to validate that this Rails magic works as intended. But I hesitated because we are not deleting address anywhere in the code. So if it's not a requirement, if it's not a *specification* to delete an address, why should it be added to the specs? I'll still do it for code coverage.

Reply:
cool, thanks!
imo, if it's worth a change in the code, it's worth a test. the only case I would skip the test is if it is too difficult to test for the value it adds, which is not the case. this is only my opinion :-)